### ISSUE TITLE
Semi-Supervised Deep Learning with Memory

### ISSUE THREAD

**Issue by aljosaosep** [aljosaosep/cool_papers#3]
**Semi-Supervised Deep Learning with Memory**
_Venue:_ ECCV'18
_Authors:_ Yanbei Chen, Xiatian Zhu, Shaogang Gong
_Keywords:_ Semi-supervised learning, memory networks, multi-class classification
_Evaluation:_ SVHN, CIFAR10, CIFAR100
_Baselines:_ Mean Teacher. Temporal Ensembling and other SOTA in semi-supervised learning
_Tasks:_ Multi-class classification, train using sparsely-labeled data.
_Brief:_ Typical classification pipeline using cross-entropy loss function, trained using label supervision, add a memory module and a (non-supervised) memory loss. 
SSL achieved via assimilation-accomodation steps, executed in every training iteration.
Use multi-slot key-value memory module. 
Key is a feature representation of each class in the feat. space. 
Update memory using labeled samples.
_Contributions:_ Use memory for SSL. In addition to (supervised) classif. loss, use (unsupervised) memory loss.

Cool idea, seems to achieve SOTA.

![image](https://user-images.githubusercontent.com/10114300/45694805-5ad58f80-bb60-11e8-8aeb-ae790ef2f301.png)
![image](https://user-images.githubusercontent.com/10114300/45694872-7e98d580-bb60-11e8-864f-df8fde019b5a.png)

![image](https://user-images.githubusercontent.com/10114300/45694918-97a18680-bb60-11e8-9d57-2db67e5503c9.png)
![image](https://user-images.githubusercontent.com/10114300/45694932-a2f4b200-bb60-11e8-8009-a9f91f955269.png)

![image](https://user-images.githubusercontent.com/10114300/45694965-b9027280-bb60-11e8-8de7-d57586f22a96.png)

**Comment by EmmaW8** [comment]
is the code released?
